src/backends/github/API.js

```diff
@@ -507,44 +437,37 @@ export default class API {
         content: contentBase64,
         encoding: "base64",
       }),
-    }).then((response) => {
-      item.sha = response.sha;
-      item.uploaded = true;
-      return item;
+    })).then(response => Object.assign({}, item, {
+      sha: response.sha,
+      uploaded: true,
     }));
   }
 
+  isFile(obj) {
+    return obj.file;
+  }
+
   updateTree(sha, path, fileTree) {
```

Comment:
I'm currently having a problem with `updateTree` where sometimes it will request a tree update, but with `tree` set to `[{}]` (empty). I'm leaving this note here as a block until I can figure out what the error is.